### ISSUE TITLE
DHCP proxy doesn't honor host config

### ISSUE THREAD

**Issue by Aetylus** [containers/netavark#695]
For an interface `eth0` with DHCP client settings configured similar to the following:
```sh
$ sudo nmcli c modify eth0 ipv4.dhcp-vendor-class-identifier my-class-1
```
and a `macvlan` Podman network with DHCP proxy:
```sh
$ sudo podman network create -d macvlan -o parent=eth0 webnetwork
$ sudo systemctl enable --now netavark-dhcp-proxy.socket
```
I would expect that the `macvlan` network honor the configuration of the parent connection through the DHCP proxy, but this doesn't appear to be the case. Testing with a simple container such as:
```sh
$ sudo podman run -dt --name webserver --network webnetwork --rm quay.io/libpod/banner
```
and using `dhcpdump -i eth0` on the host, I'm not seeing the DHCP option being sent in requests by the container. I have confirmed the host will send it if it makes a DHCP request itself however.

Is this intended behavior? Is there any way to get the network to use the host's DHCP configuration?

**Comment by Luap99** [comment]
> Is this intended behavior? 

Yes the proxy has no idea about any settings on the host interface. First of all how would we even know what dhcp settings do you use? We have no idea until we start reading NetworkManager configs but then other users may not be using network manager so how can we even know that. Supporting this would be a nightmare.
Second, while vendor class my be fine to send for more than once for multiple request if something like dhcp-client-id is set it would not work at all.

> Is there any way to get the network to use the host's DHCP configuration?

Not at present. The only way I see us supporting this is to allow some options as network options, i.e. `podman network create --opt ...`

**Comment by Luap99** [comment]
Is there a specific reason that you have to set dhcp-vendor-class-identifier? We can consider adding this as feature if you need it.

**Comment by Aetylus** [comment]
> Is there a specific reason that you have to set dhcp-vendor-class-identifier? We can consider adding this as feature if you need it.

I was looking to configure a separate set of DHCP options (through `dnsmasq` tags or `dhcpd` classes) for Podman containers and was hoping to see if I could identify them using the vendor class identifier (as that seemed the most appropriate DHCP option available). I'm not sure how useful it would be for others but if it's possibly helpful down the line for others, having it available as a feature would be nice!
